### PR TITLE
feat: add toggle password to password fields

### LINKED ISSUE #384: Add toggle password button to all password fields
You can follow [this](https://github.com/joinum/parzival/pull/323) PR.
Also, don't forget to create a component and reuse it 💡

### PR DESCRIPTION
**Pull request by MarioRodrigues10**

### CODE REVIEW

File: lib/atomic_web/templates/user_session/new.html.heex [hunk 1]
```diff
@@ -14,23 +14,29 @@
...
         <%= label(f, :email, class: "sr-only") %>
         <%= email_input(f, :email,
           required: true,
           placeholder: gettext("Email address"),
-          class: "relative w-full sm:w-96 mx-4 appearance-none rounded border border-zinc-300 px-3 py-2 text-zinc-900 placeholder-zinc-500 focus:z-10 focus:border-orange-400 focus:ring-orange-400 focus:outline-none sm:text-sm"
+          class: "relative w-96 appearance-none rounded border border-zinc-300 text-zinc-900 placeholder-zinc-500 focus:z-10 focus:border-orange-400 focus:ring-orange-400 focus:outline-none sm:text-sm"
         ) %>
       </div>
-      <div class="flex justify-center">
+      <div x-data="{ hide: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:outline-none sm:text-sm">
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_session/new.html.heex]:
```suggestion
      <div x-data="{ hide: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:ring-1 sm:text-sm">

```

File: lib/atomic_web/templates/user_session/new.html.heex [hunk 2]
```diff
@@ -14,23 +14,29 @@
...
           autocomplete: "current-password",
-          class: "relative w-full sm:w-96 mx-4 appearance-none rounded border border-zinc-300 px-3 py-2 text-zinc-900 placeholder-zinc-500 focus:z-10 focus:border-orange-400 focus:ring-orange-400 focus:outline-none sm:text-sm"
+          class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:ring-0"
         ) %>
+        <div x-show="hide" x-on:click="hide = false; document.getElementById('user_password').type = 'text';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
+        <div x-show="!hide" x-on:click="hide = true; document.getElementById('user_password').type = 'password';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye_off class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_session/new.html.heex]:
```suggestion
          class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:ring-0 sm:text-sm"
        ) %>
        <div x-show="hide" x-on:click="hide = false; document.getElementById('user_password').type = 'text';" class="px-2 w-fit cursor-pointer">
          <Ionicons.Filled.eye class="w-5 h-auto fill-zinc-900 hover:fill-black" />
        </div>
        <div x-show="!hide" x-on:click="hide = true; document.getElementById('user_password').type = 'password';" class="px-2 w-fit cursor-pointer">
          <Ionicons.Filled.eye_off class="w-5 h-auto fill-zinc-900 hover:fill-black" />
        </div>
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 3]
```diff
@@ -14,36 +14,48 @@
...
         ) %>
         <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :email) %></span>
       </div>
-      <div class="flex flex-col items-center justify-center mx-4">
+      <div x-data="{ hide: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:outline-none sm:text-sm">
         <%= label(f, :password, class: "sr-only") %>
         <%= password_input(f, :password,
           required: true,
           placeholder: gettext("Password"),
           autocomplete: "current-password",
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
      <div class="flex flex-col items-center justify-center">
        <div x-data="{ hide: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:ring-1 sm:text-sm">
          <%= label(f, :password, class: "sr-only") %>
          <%= password_input(f, :password,
            required: true,
            placeholder: gettext("Password"),
            autocomplete: "current-password",
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 4]
```diff
@@ -14,36 +14,48 @@
...
-          class: "relative w-full sm:w-96 mx-4 appearance-none rounded border border-zinc-300 px-3 py-2 text-zinc-900 placeholder-zinc-500 focus:z-10 focus:border-orange-400 focus:ring-orange-400 focus:outline-none sm:text-sm"
+          class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:outline-none focus:ring-0"
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
          class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:outline-none focus:ring-0 sm:text-sm"
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 5]
```diff
@@ -14,36 +14,48 @@
...
         ) %>
-        <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password) %></span>
+        <div x-show="hide" x-on:click="hide = false; document.getElementById('user_password').type = 'text';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
+        <div x-show="!hide" x-on:click="hide = true; document.getElementById('user_password').type = 'password';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye_off class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
        <div x-show="hide" x-on:click="hide = false; document.getElementById('user_password').type = 'text';" class="px-2 w-fit cursor-pointer">
            <Ionicons.Filled.eye class="w-5 h-auto fill-black hover:fill-black" />
          </div>
          <div x-show="!hide" x-on:click="hide = true; document.getElementById('user_password').type = 'password';" class="px-2 w-fit cursor-pointer">
            <Ionicons.Filled.eye_off class="w-5 h-auto fill-black hover:fill-black" />
          </div>
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 6]
```diff
@@ -14,36 +14,48 @@
...
       </div>
-      <div class="flex flex-col items-center justify-center mx-4">
+      <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password) %></span>
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
        <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password) %></span>
      </div>
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 7]
```diff
@@ -14,36 +14,48 @@
...
+      <div x-data="{ hide_confirm: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:outline-none  sm:text-sm">
         <%= label(f, :password_confirmation, class: "sr-only") %>
         <%= password_input(f, :password_confirmation,
           required: true,
           placeholder: gettext("Confirm password"),
           autocomplete: "current-password",
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
      <div class="flex flex-col items-center justify-center">
        <div x-data="{ hide_confirm: true }" class="flex flex-row items-center justify-center w-full block border border-zinc-300 rounded placeholder-zinc-500 focus-within:z-10 focus-within:border-orange-400 focus-within:ring-orange-400 focus-within:ring-1  sm:text-sm">
          <%= label(f, :password_confirmation, class: "sr-only") %>
          <%= password_input(f, :password_confirmation,
            required: true,
            placeholder: gettext("Confirm password"),
            autocomplete: "current-password",
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 8]
```diff
@@ -14,36 +14,48 @@
...
-          class: "relative w-full sm:w-96 mx-4 appearance-none rounded border border-zinc-300 px-3 py-2 text-zinc-900 placeholder-zinc-500 focus:z-10 focus:border-orange-400 focus:ring-orange-400 focus:outline-none sm:text-sm"
+          class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:outline-none focus:ring-0"
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
                      class: "relative w-full border-0 text-zinc-900 rounded appearance-none outline-none focus:outline-none focus:ring-0 sm:text-sm"
```

File: lib/atomic_web/templates/user_registration/new.html.heex [hunk 9]
```diff
@@ -14,36 +14,48 @@
...
-        <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password_confirmation) %></span>
+        <div x-show="hide_confirm" x-on:click="hide_confirm = false; document.getElementById('user_password_confirmation').type = 'text';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
+        <div x-show="!hide_confirm" x-on:click="hide_confirm = true; document.getElementById('user_password_confirmation').type = 'password';" class="px-2 w-fit cursor-pointer">
+          <Ionicons.Filled.eye_off class="w-5 h-auto fill-black hover:fill-black" />
+        </div>
       </div>
+      <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password_confirmation) %></span>
     </div>
```
Comment (joaodiaslobo) [inline lib/atomic_web/templates/user_registration/new.html.heex]:
```suggestion
                  <div x-show="hide_confirm" x-on:click="hide_confirm = false; document.getElementById('user_password_confirmation').type = 'text';" class="px-2 w-fit cursor-pointer">
            <Ionicons.Filled.eye class="w-5 h-auto fill-black hover:fill-black" />
          </div>
          <div x-show="!hide_confirm" x-on:click="hide_confirm = true; document.getElementById('user_password_confirmation').type = 'password';" class="px-2 w-fit cursor-pointer">
            <Ionicons.Filled.eye_off class="w-5 h-auto fill-black hover:fill-black" />
          </div>
        </div>
        <span class="text-sm w-full sm:w-96 mx-4 text-red-600"><%= error_tag(f, :password_confirmation) %></span>
      </div>
    </div>
```

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
Change password inputs to have the same text size as the other inputs and same ring size on focus.  Fix the registration page password inputs to have the same spacing.

Also, the mobile view's elements are not sized properly:

image

**Review by ruilopesm — COMMENTED** [review]
Both inputs should occupy the same width as the "Log in" button. Just like it was before 🙏🏼

**Comment by joaodiaslobo** [comment]
Also, can you add the following CSS code to `app.css` to disable the toggle Microsoft Edge adds to show the password?

```css
/* Hide password reveal button in MS Edge */
::-ms-reveal {
  display: none;
}
```

To avoid this:

image

**Review by joaodiaslobo — APPROVED** [review]
Good job! 🚀

**Pull request merged**